packages/services/f-feature-management/src/FeatureManager.js

```diff
@@ -0,0 +1,123 @@
+import evaluateLogic from './evaluateLogic';
+import evaluateDefault from './evaluateDefaultValue';
+import { getFeature } from './featureGetter';
+import { getContext } from './contextGetter';
+import trackExperiment from './trackExperiment';
+import { logger } from './logger';
+
+
+const logValueRequest = (func, key, value) => {
+    logger.logInfo(`${func.name}: ${key} => ${value}`);
+};
+
+/**
+ * Class used to access features.  Should be created using `createFeatureManagement`
+ */
+class FeatureManager {
+    /**
+     *
+     * @param {object} settings
+     */
+    constructor (settings) {
+        this._keyPrefix = settings.keyPrefix;
+    }
+
+    /**
+     * Returns the full feature key, including any prefix
+     * @param {string} key
+     * @returns Full feature key
+     */
+    createFullKey (key) {
+        return this._keyPrefix
+            ? `${this._keyPrefix}::${key}`
+            : key;
+    }
+
+    /**
+     * Gets the value of the feature, and tracks any experiments.
+     * @param {string} key
+     * @returns Value of feature
+     */
+    getValue (key) {
+        logValueRequest(this.getValue, key, null);
+
+        try {
+            const feature = getFeature(this.createFullKey(key));
+
+            if (!feature) {
+                return null;
+            }
+
+            const context = getContext();
+
+            const variant = evaluateLogic(feature, context);
+
+            if (variant) {
+                if (variant.experimentKey && variant.experimentVariant) {
+                    trackExperiment(variant);
+                }
+
+                return variant.value;
+            }
+
+            const defaultValue = evaluateDefault(feature, context);
+
+            return defaultValue;
+        } catch (error) {
+            logger.logError(error.message, null, { error });
+            return null;
+        }
+    }
+
+    /**
+     * Gets feature and tracks any experiments.  Checks it's a boolean before returning.
+     * @param {string} key
+     * @returns Boolean or null
+     */
+    getBooleanValue (key) {
```

Comment:
Is the developer going to know what type to retrieve?

Reply:
We shall see!  I actually think they'll normally just call `getValue()`.

Reply:
If you have an opinion the typed ones being counterproductive please yell and we could look at removing them.

Reply:
Its a bit hard to say at the moment as I can't picture the whole thing in my head very clearly yet; I guess if there is a value in having the specific typings then its cool. Just trying to think as a consumer would I actually now which one to call

Reply:
I'm not sure.  It's not a very javascripty way to think.  @jnewc put them in in the first place so may have an opinion.  In any case no consumers yet so we can change before it gets to primetime.

Reply:
We actually discussed it at the time https://github.com/justeat/fozzie-components/pull/1097#discussion_r664699137